src/umpire/device_allocator_helper.cpp

```diff
@@ -0,0 +1,119 @@
+////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC and Umpire
+// project contributors. See the COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (MIT)
+////////////////////////////////////////////////////////////////////////////
+#include "umpire/device_allocator_helper.hpp"
+
+#include <string.h>
+
+#include "umpire/ResourceManager.hpp"
+#include "umpire/alloc/CudaMallocManagedAllocator.hpp"
+
+namespace umpire {
+
+//////////////////////////////////////////////////////////////////////////
+// Global variables for host and device
+//////////////////////////////////////////////////////////////////////////
+__managed__ int UMPIRE_TOTAL_DEV_ALLOCS{10};
+
+DeviceAllocator* UMPIRE_DEV_ALLOCS_h{nullptr};
+__device__ DeviceAllocator* UMPIRE_DEV_ALLOCS{nullptr};
+
+//////////////////////////////////////////////////////////////////////////
+// host/device functions
+//////////////////////////////////////////////////////////////////////////
+__host__ __device__ DeviceAllocator getDeviceAllocator(const char* name)
+{
+  int index = findDeviceAllocatorID(name);
+  if (index == -1)
+    UMPIRE_ERROR("No DeviceAllocator by the name " << name << " was found.");
+#if !defined(__CUDA_ARCH__)
+  return UMPIRE_DEV_ALLOCS_h[index];
+#else
+  return UMPIRE_DEV_ALLOCS[index];
+#endif
+}
+
+__host__ __device__ DeviceAllocator getDeviceAllocator(int id)
+{
+  if (id < 0 || id > UMPIRE_TOTAL_DEV_ALLOCS)
+    UMPIRE_ERROR("Invalid ID given.");
+  if (!deviceAllocatorExists(id))
+    UMPIRE_ERROR("No DeviceAllocator by with that ID was found.");
+
+#if !defined(__CUDA_ARCH__)
+  return UMPIRE_DEV_ALLOCS_h[id];
+#else
+  return UMPIRE_DEV_ALLOCS[id];
+#endif
+}
+
+__host__ __device__ int findDeviceAllocatorID(const char* name)
+{
+#if !defined(__CUDA_ARCH__)
+  for (int i = 0; i < UMPIRE_TOTAL_DEV_ALLOCS; i++) {
+    if (strcmp(UMPIRE_DEV_ALLOCS_h[i].getName(), name) == 0)
+      return i;
+  }
+  return -1;
+#else
+  for (int i = 0; i < UMPIRE_TOTAL_DEV_ALLOCS; i++) {
+    const char* temp = UMPIRE_DEV_ALLOCS[i].getName();
+    int index = 0;
+    int tally = 0;
+    do {
+      if (temp[index] == 0)
+        break;
+      if (temp[index] != name[index])
+        tally++;
+    } while (name[index++] != 0);
+    if (tally == 0)
+      return i;
+  }
+  return -1;
+#endif
+}
+
+__host__ __device__ bool deviceAllocatorExists(int id)
```

Comment:
I think this should be called `isDeviceAllocator` to match our other functions.

Reply:
Good idea